submissions/Nemooochka/tiny-js-upd/index.js

```diff
@@ -0,0 +1,121 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(species, name, gender, saying, friends) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.friends = friends;
+    }
+
+    sayHi() {
+        return( `${this.saying} I'm a ${this.species} with the name ${this.name}, which means I'm ${this.gender}. I want to say hi to <b>${this.friends}</b>, you are the best!`);
+    }
+}
+
+class Animal extends Inhabitant {
+    legs = 4;
+    constructor(name, gender, friends) {
+        super();
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+    }
+
+    sayHi() {
+        return(
+            super.sayHi() + ` By the way, I have ${this.legs} legs.`
+        )
+    }
+}
+
+
+class Dog extends Animal {
+    species = 'dog';
+    saying = 'Woof-woof';
+    constructor(name, gender, friends) {
+        super(name, gender, friends);
+    }
+}
+
+class Cat extends Animal {
+    species = 'cat';
+    saying = 'Meeeow!';
+    constructor(name, gender, friends) {
+        super(name, gender, friends);
+    }
+}
+
+class Human extends Inhabitant {
+    species = 'human';
+    hands = 2;
+    constructor(name, friends) {
+        super();
+        this.name = name;
+        this.friends = friends;
+    }
+
+    sayHi() {
+        return(
+            super.sayHi() + ` By the way, I have ${this.hands} hands.`
+        )
+    }
+}
+
+class Man extends Human {
+    gender = 'male';
+    saying = 'How you doing?';
```

Comment:
So, all men will have the same saying. In real life humans are way more customizable compared with animals.
Please allow humans to have individual greetings.